src/Bridge/Graphql/Resolver/ItemResolverFactory.php

```diff
@@ -0,0 +1,87 @@
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Graphql\Resolver;
+
+use ApiPlatform\Core\Api\IdentifiersExtractorInterface;
+use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
+use ApiPlatform\Core\DataProvider\SubresourceDataProviderInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use GraphQL\Type\Definition\ResolveInfo;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Creates a function retrieving an item to resolve a GraphQL query.
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ *
+ * @internal
+ */
+final class ItemResolverFactory extends AbstractResolverFactory implements ItemResolverFactoryInterface
+{
+    private $itemDataProvider;
+    private $normalizer;
+    private $identifiersExtractor;
+
+    public function __construct(ItemDataProviderInterface $itemDataProvider, SubresourceDataProviderInterface $subresourceDataProvider, NormalizerInterface $normalizer, IdentifiersExtractorInterface $identifiersExtractor)
+    {
+        parent::__construct($subresourceDataProvider);
+
+        $this->itemDataProvider = $itemDataProvider;
+        $this->normalizer = $normalizer;
+        $this->identifiersExtractor = $identifiersExtractor;
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function createItemResolver(string $resourceClass, string $rootClass): callable
+    {
+        return function ($root, $args, $context, ResolveInfo $info) use ($resourceClass, $rootClass) {
+            $rootProperty = $info->fieldName;
+            $rootIdentifiers = $this->identifiersExtractor->getIdentifiersFromResourceClass($rootClass);
+            if (isset($root[$rootProperty])) {
+                $item = $this->getSubresource($rootClass, $root, $rootIdentifiers, $rootProperty, $resourceClass, false);
+            } else {
+                $identifiers = [];
+                $uniqueIdentifier = [];
+                foreach ($rootIdentifiers as $rootIdentifier) {
+                    if (!isset($args[$rootIdentifier])) {
+                        continue;
+                    }
+
+                    if (\is_array($args[$rootIdentifier])) {
+                        if (\count($args[$rootIdentifier]) > 1) {
+                            throw new InvalidArgumentException('Composite identifiers are not allowed for a resource already used as a composite identifier');
+                        }
+
+                        $identifiers[] = $rootIdentifier.'='.current($args[$rootIdentifier]);
+
+                        continue;
+                    }
+
+                    $identifiers[] = $rootIdentifier.'='.$args[$rootIdentifier];
+                    $uniqueIdentifier[] = $args[$rootIdentifier];
+                }
+
+                if (!$identifiers) {
+                    return null;
+                }
+
+                $item = $this->itemDataProvider->getItem($resourceClass, \count($identifiers) > 1 ? implode(';', $identifiers) : $uniqueIdentifier[0]);
```

Comment:
Just a note to myself but this would need refactoring if we can implement some kind of identifiers denormalization (https://github.com/api-platform/core/issues/1309) not linked to any bridge (code duplication).